dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AccountController.java

```diff
@@ -190,37 +191,32 @@ private void handleRecoveryLock( String username )
     }
 
     @PostMapping( "/restore" )
-    public void restoreAccount(
-        @RequestParam String token,
-        @RequestParam String password,
-        HttpServletRequest request,
-        HttpServletResponse response )
-        throws WebMessageException
+    @ResponseBody
+    public WebMessage restoreAccount( @RequestParam String token, @RequestParam String password )
     {
         String[] idAndRestoreToken = securityService.decodeEncodedTokens( token );
         String idToken = idAndRestoreToken[0];
-        String restoreToken = idAndRestoreToken[1];
 
         UserCredentials credentials = userService.getUserCredentialsByIdToken( idToken );
 
-        if ( credentials == null )
+        if ( credentials == null || idAndRestoreToken.length < 2 )
```

Comment:
FYI: adding `|| idAndRestoreToken.length < 2` is a fix for index out of bounds exception when calling with illegal `token`.